src/OrleansRuntime/Catalog/DefaultGrainActivator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// <see cref="IGrainActivator"/> that uses type activation to create grains.
+    /// </summary>
+    public class DefaultGrainActivator : IGrainActivator
```

Comment:
GrainFacory (or GrainActivationFactory) would be more appropriate to the abstraction that will replace our current GrainCreator, as it matches [DefaultControllerFactory](https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/Controllers/DefaultControllerFactory.cs), which in our case can have stuff like the facet system we are planning in the future (MVC does a VERY similar thing to our facets).
Maybe `IGrainActivationCreator`? I don't know, none seem right, but at least IGrainActivator matches MVC naming, so we at least had familiarity on our side, but I'm definitely open to change this